Move colon from beginning to end

themaskedcrusader opened the pull request:
Since the FILESEXTRAPATHS_append variable modifies the search path for src_uris, the deliminator colon should be at the end per the yocto documentation: https://www.yoctoproject.org/docs/2.4.3/ref-manual/ref-manual.html#var-FILESEXTRAPATHS

With the colon at the end, other recipes can bbappend to this recipe. without a colon at the end  (per the convention), the other recipe's path is appended to this recipe's path, causing errors during bitbake's do_fetch

shr-project commented:
What error you're seeing without this? Yes, most FILESEXTRAPATHS have colon at the end, but they are usually used with _prepend not _append used in this case.

So your version makes it worse I believe.

themaskedcrusader commented:
I added a recipe qtbase_%.bbappend because I needed to add a mutex to QT to resolve an issue. I added the following lines to my bbappend:

`FILESEXTRAPATHS_append := "${THISDIR}/files"

SRC_URI_append = " \
    file://file.patch \
"`

but when I run bitbake I get the following error:

`WARNING: qtbase-5.10.1+gitAUTOINC+6c6ace9d23-r0 do_fetch: Failed to fetch URL file://run-ptest, attempting MIRRORS if available
ERROR: qtbase-5.10.1+gitAUTOINC+6c6ace9d23-r0 do_fetch: Fetcher failure: Unable to find file file://run-ptest anywhere. The paths that were searched were:
    /yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/genesis-imx
    /yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/genesis-imx
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/genesis-imx
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/genesis-imx
    /yocto/sources/meta-qt5/recipes-qt/qt5/files/genesis-imx
    /yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/aarch64
    /yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/aarch64
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/aarch64
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/aarch64
    /yocto/sources/meta-qt5/recipes-qt/qt5/files/aarch64
    /yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/genesis-imx8mmevk
    /yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/genesis-imx8mmevk
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/genesis-imx8mmevk
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/genesis-imx8mmevk
    /yocto/sources/meta-qt5/recipes-qt/qt5/files/genesis-imx8mmevk
    /yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/mx8mm
    /yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/mx8mm
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/mx8mm
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/mx8mm
    /yocto/sources/meta-qt5/recipes-qt/qt5/files/mx8mm
    /yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/imxgpu3d
    /yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/imxgpu3d
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/imxgpu3d
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/imxgpu3d
    /yocto/sources/meta-qt5/recipes-qt/qt5/files/imxgpu3d
    /yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/imxgpu2d
    /yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/imxgpu2d
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/imxgpu2d
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/imxgpu2d
    /yocto/sources/meta-qt5/recipes-qt/qt5/files/imxgpu2d
    /yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/imxgpu
    /yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/imxgpu
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/imxgpu
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/imxgpu
    /yocto/sources/meta-qt5/recipes-qt/qt5/files/imxgpu
    /yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/imxvpuhantro
    /yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/imxvpuhantro
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/imxvpuhantro
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/imxvpuhantro
    /yocto/sources/meta-qt5/recipes-qt/qt5/files/imxvpuhantro
    /yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/imxvpu
    /yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/imxvpu
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/imxvpu
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/imxvpu
    /yocto/sources/meta-qt5/recipes-qt/qt5/files/imxvpu
    /yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/imxdrm
    /yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/imxdrm
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/imxdrm
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/imxdrm
    /yocto/sources/meta-qt5/recipes-qt/qt5/files/imxdrm
    /yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/mx8m
    /yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/mx8m
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/mx8m
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/mx8m
    /yocto/sources/meta-qt5/recipes-qt/qt5/files/mx8m
    /yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/mx8
    /yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/mx8
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/mx8
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/mx8
    /yocto/sources/meta-qt5/recipes-qt/qt5/files/mx8
    /yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/imx
    /yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/imx
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/imx
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/imx
    /yocto/sources/meta-qt5/recipes-qt/qt5/files/imx
    /yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/aarch64
    /yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/aarch64
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/aarch64
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/aarch64
    /yocto/sources/meta-qt5/recipes-qt/qt5/files/aarch64
    /yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/
    /yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/
    /yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/
    /yocto/sources/meta-qt5/recipes-qt/qt5/files/
    /yocto/build/../downloads
ERROR: qtbase-5.10.1+gitAUTOINC+6c6ace9d23-r0 do_fetch: Fetcher failure for URL: 'file://run-ptest'. Unable to fetch URL from any source.
ERROR: qtbase-5.10.1+gitAUTOINC+6c6ace9d23-r0 do_fetch: Function failed: base_do_fetch
ERROR: Logfile of failure stored in: /yocto/build/tmp/work/aarch64-mx8mm-poky-linux/qtbase/5.10.1+gitAUTOINC+6c6ace9d23-r0/temp/log.do_fetch.28769
ERROR: Task (/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase_git.bb:do_fetch) failed with exit code '1'
`

Since the documentation from yoctoproject always has the colon at the end of the variable value, every recipe should follow the same issue. You'll see the error in the line ** /yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/**

It's concatenating my path to the ptest path, which highlights this error. 

Your own recipe "https://github.com/meta-qt5/meta-qt5/blob/master/recipes-qt/qt5/nativesdk-qtbase_git.bb" has the variable with the colon at the end, and the following public recipes all show compliance with the documentation by having the colon at the end:

- https://github.com/openembedded/meta-openembedded/blob/master/meta-python/recipes-devtools/python/python-gevent.inc
- http://git.yoctoproject.org/cgit/cgit.cgi/poky/tree/meta-poky/recipes-core/psplash/psplash_git.bbappend
- and many others

Again, I refer you to the documentation that shows the colon to be at the end to terminate the path or delimit the string so another string can be added.

Look, I don't mind adding a colon to both sides of the variable in my bbappend file, but it caused me a tiny headache to find the cause why my recipe was broken that I'm just trying to help out the community.

themaskedcrusader commented:
I added a recipe qtbase_%.bbappend because I needed to add a mutex to QT to resolve an issue. I added the following lines to my bbappend:

FILESEXTRAPATHS_append := "${THISDIR}/files"

SRC_URI_append = " 
file://file.patch 
"

but when I run bitbake I get the following error:

WARNING: qtbase-5.10.1+gitAUTOINC+6c6ace9d23-r0 do_fetch: Failed to fetch URL file://run-ptest, attempting MIRRORS if available
ERROR: qtbase-5.10.1+gitAUTOINC+6c6ace9d23-r0 do_fetch: Fetcher failure: Unable to find file file://run-ptest anywhere. The paths that were searched were:
/yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/distro-imx
/yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/distro-imx
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/distro-imx
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/distro-imx
/yocto/sources/meta-qt5/recipes-qt/qt5/files/distro-imx
/yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/aarch64
/yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/aarch64
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/aarch64
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/aarch64
/yocto/sources/meta-qt5/recipes-qt/qt5/files/aarch64
/yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/distro-imx8mmevk
/yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/distro-imx8mmevk
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/distro-imx8mmevk
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/distro-imx8mmevk
/yocto/sources/meta-qt5/recipes-qt/qt5/files/distro-imx8mmevk
/yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/mx8mm
/yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/mx8mm
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/mx8mm
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/mx8mm
/yocto/sources/meta-qt5/recipes-qt/qt5/files/mx8mm
/yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/imxgpu3d
/yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/imxgpu3d
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/imxgpu3d
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/imxgpu3d
/yocto/sources/meta-qt5/recipes-qt/qt5/files/imxgpu3d
/yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/imxgpu2d
/yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/imxgpu2d
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/imxgpu2d
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/imxgpu2d
/yocto/sources/meta-qt5/recipes-qt/qt5/files/imxgpu2d
/yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/imxgpu
/yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/imxgpu
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/imxgpu
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/imxgpu
/yocto/sources/meta-qt5/recipes-qt/qt5/files/imxgpu
/yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/imxvpuhantro
/yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/imxvpuhantro
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/imxvpuhantro
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/imxvpuhantro
/yocto/sources/meta-qt5/recipes-qt/qt5/files/imxvpuhantro
/yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/imxvpu
/yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/imxvpu
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/imxvpu
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/imxvpu
/yocto/sources/meta-qt5/recipes-qt/qt5/files/imxvpu
/yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/imxdrm
/yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/imxdrm
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/imxdrm
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/imxdrm
/yocto/sources/meta-qt5/recipes-qt/qt5/files/imxdrm
/yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/mx8m
/yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/mx8m
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/mx8m
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/mx8m
/yocto/sources/meta-qt5/recipes-qt/qt5/files/mx8m
/yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/mx8
/yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/mx8
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/mx8
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/mx8
/yocto/sources/meta-qt5/recipes-qt/qt5/files/mx8
/yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/imx
/yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/imx
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/imx
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/imx
/yocto/sources/meta-qt5/recipes-qt/qt5/files/imx
/yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/aarch64
/yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/aarch64
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/aarch64
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/aarch64
/yocto/sources/meta-qt5/recipes-qt/qt5/files/aarch64
/yocto/sources/meta-fsl-bsp-release/imx/meta-sdk/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase/
/yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase-5.10.1+gitAUTOINC+6c6ace9d23/
/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase/
/yocto/sources/meta-qt5/recipes-qt/qt5/files/
/yocto/build/../downloads
ERROR: qtbase-5.10.1+gitAUTOINC+6c6ace9d23-r0 do_fetch: Fetcher failure for URL: 'file://run-ptest'. Unable to fetch URL from any source.
ERROR: qtbase-5.10.1+gitAUTOINC+6c6ace9d23-r0 do_fetch: Function failed: base_do_fetch
ERROR: Logfile of failure stored in: /yocto/build/tmp/work/aarch64-mx8mm-poky-linux/qtbase/5.10.1+gitAUTOINC+6c6ace9d23-r0/temp/log.do_fetch.28769
ERROR: Task (/yocto/sources/meta-qt5/recipes-qt/qt5/qtbase_git.bb:do_fetch) failed with exit code '1'

Since the documentation from yoctoproject always has the colon at the end of the variable value, every recipe should follow the same issue. You'll see the error in the line ** /yocto/sources/meta-qt5/recipes-qt/qt5/ptest/yocto/sources/meta-mylayer/recipes-qt/qt5/qtbase/**

It's concatenating my path to the ptest path, which highlights this error.

Your own recipe "https://github.com/meta-qt5/meta-qt5/blob/master/recipes-qt/qt5/nativesdk-qtbase_git.bb" has the variable with the colon at the end, and the following public recipes all show compliance with the documentation by having the colon at the end:

https://github.com/openembedded/meta-openembedded/blob/master/meta-python/recipes-devtools/python/python-gevent.inc
http://git.yoctoproject.org/cgit/cgit.cgi/poky/tree/meta-poky/recipes-core/psplash/psplash_git.bbappend
and many others
Again, I refer you to the documentation that shows the colon to be at the end to terminate the path or delimit the string so another string can be added.

Look, I don't mind adding a colon to both sides of the variable in my bbappend file, but it caused me a tiny headache to find the cause why my recipe was broken that I'm just trying to help out the community.

themaskedcrusader commented:
prepend and append should synonymous, since the colon is added to the end without _prepend or _append either. if the variable is always terminated with a colon, then the path will always properly be delimited.

shr-project commented:
Read the manual to understand what is the difference between prepend and append.

Separator needs to be at the end when prepending and at the beginning when appending.

Fix your bbappend.
